webapp/client/src/pages/TaxNumberPage.js

```diff
@@ -0,0 +1,331 @@
+import PropTypes from "prop-types";
+import React, { useState } from "react";
+import { Trans, useTranslation } from "react-i18next";
+import FormFieldConsentBox from "../components/FormFieldConsentBox";
+import FormFieldDropDown from "../components/FormFieldDropDown";
+import FormFieldYesNo from "../components/FormFieldYesNo";
+import FormFieldTaxNumber from "../components/FormFieldTaxNumber";
+import FormHeader from "../components/FormHeader";
+import FormRowCentered from "../components/FormRowCentered";
+import StepForm from "../components/StepForm";
+import StepHeaderButtons from "../components/StepHeaderButtons";
+import SubHeading from "../components/SubHeading";
+
+const TAX_NUMBER_FORM_STATES = {
+  NothingSelected: "NothingSelected",
+  StateSelectionTaxNumberExists: "StateSelectionTaxNumberExists",
+  StateSelectionNoTaxNumberExists: "StateSelectionNoTaxNumberExists",
+  TaxOfficeSelection: "TaxOfficeSelection",
+  TaxNumberInput: "TaxNumberInput",
+  RequestNewTaxNumber: "RequestNewTaxNumber",
+};
+
+function currentState(taxNumberFormState, fields) {
+  if (taxNumberFormState === TAX_NUMBER_FORM_STATES.NothingSelected) {
+    if (fields.steuernummerExists.value === "yes") {
+      if (fields.bundesland.selectedValue) {
+        return TAX_NUMBER_FORM_STATES.TaxNumberInput;
+      }
+      return TAX_NUMBER_FORM_STATES.StateSelectionTaxNumberExists;
+    }
+    if (fields.steuernummerExists.value === "no") {
+      if (fields.bundesland.selectedValue) {
+        if (fields.bufaNr.selectedValue) {
+          return TAX_NUMBER_FORM_STATES.RequestNewTaxNumber;
+        }
+        return TAX_NUMBER_FORM_STATES.TaxOfficeSelection;
+      }
+      return TAX_NUMBER_FORM_STATES.StateSelectionNoTaxNumberExists;
+    }
+  }
+
+  return taxNumberFormState;
+}
```

Comment:
I would add the following comment. Would that help?:
`This function is used to set the correct state for the tax number page through the provided props. It will only change the state if no state has been set.`

That function is called on every render of the TaxNumberPage (s. line 238). The reason for this function is that we do need to change the set through the props only if there has nothing been selected.

However, we could also get rid of the if and use the function like this in line 75:
```javascript
const [taxNumberFormState, setTaxNumberFormState] = useState(
    currentState(fields)
  );
```

Wdyth?

Reply:
Uh, I like the suggestion a lot. 👌